Flaky integration tests with short timeout

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
If the matter is security sensitive, please disclose it privately to a security contact: https://github.com/kudobuilder/kudo/blob/master/SECURITY_CONTACTS
-->


**What happened**:

The integration tests failed while the plan was still in progress:

```
        --- FAIL: kudo/harness/immutable-client-ip (10.08s)
            logger.go:37: 19:49:44 | immutable-client-ip | Creating namespace: kudo-test-topical-redfish
            logger.go:37: 19:49:44 | immutable-client-ip/0-install-instance | starting test step 0-install-instance
            logger.go:37: 19:49:44 | immutable-client-ip/0-install-instance | Instance:kudo-test-topical-redfish/my-instance created
            logger.go:37: 19:49:44 | immutable-client-ip/0-install-instance | OperatorVersion:kudo-test-topical-redfish/service-operator created
            logger.go:37: 19:49:55 | immutable-client-ip/0-install-instance | test step failed 0-install-instance
            case.go:91: --- Instance:kudo-test-topical-redfish/my-instance
                +++ Instance:kudo-test-topical-redfish/my-instance
                @@ -1,8 +1,19 @@
                 apiVersion: kudo.k8s.io/v1alpha1
                 kind: Instance
                 metadata:
                +  labels:
                +    operator: service-operator
                   name: my-instance
                   namespace: kudo-test-topical-redfish
                +spec:
                +  operatorVersion:
                +    kind: OperatorVersion
                +    name: service-operator
                 status:
                -  status: COMPLETE
                +  activePlan:
                +    apiVersion: kudo.k8s.io/v1alpha1
                +    kind: PlanExecution
                +    name: my-instance-deploy-966623451
                +    namespace: kudo-test-topical-redfish
                +    uid: b3555e52-a282-11e9-b9ea-0242ac110002
                 
                
            case.go:91: resource Instance:kudo-test-topical-redfish/my-instance: .status.status: key is missing from map
            case.go:94: failed in step 0-install-instance
            logger.go:37: 19:49:55 | immutable-client-ip | Deleting namespace: kudo-test-topical-redfish
```

I suspect that as we add more tests (and the more restricted environment of circle ci), we will need to increase the default timeout of the tests. It is ten seconds right now, which is likely not long enough.

https://circleci.com/gh/kudobuilder/kudo/2307?utm_campaign=workflow-failed&utm_medium=email&utm_source=notification

--- a/keps/0008-operator-testing.md
+++ b/keps/0008-operator-testing.md
@@ -153,6 +153,8 @@ type TestSuite struct {
 	StartControlPlane bool
 	// Whether or not to start the KUDO controller for the tests.
 	StartKUDO         bool
+	// Override the default assertion timeout of 30 seconds (in seconds).
+	Timeout int
 }
 
 ```
@@ -295,7 +297,7 @@ When searching the assertion file for a test step, if a `TestAssert` object is f
 type TestAssert struct {
     // The type meta object, should always be a GVK of kudo.k8s.io/v1alpha1/TestAssert.
     TypeMeta
-    // Override the default timeout of 300 seconds (in seconds).
+    // Override the default timeout of 30 seconds (in seconds).
     Timeout int
 }
 ```

--- a/pkg/apis/kudo/v1alpha1/test_types.go
+++ b/pkg/apis/kudo/v1alpha1/test_types.go
@@ -26,6 +26,8 @@ type TestSuite struct {
 	StartKUDO bool `json:"startKUDO"`
 	// If set, do not delete the resources after running the tests.
 	SkipDelete bool `json:"skipDelete"`
+	// Override the default timeout of 30 seconds (in seconds).
+	Timeout int `json:"timeout"`
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
@@ -54,7 +56,7 @@ type TestStep struct {
 type TestAssert struct {
 	// The type meta object, should always be a GVK of kudo.k8s.io/v1alpha1/TestAssert.
 	metav1.TypeMeta `json:",inline"`
-	// Override the default timeout of 300 seconds (in seconds).
+	// Override the default timeout of 30 seconds (in seconds).
 	Timeout int `json:"timeout"`
 }
 

--- a/pkg/test/case.go
+++ b/pkg/test/case.go
@@ -28,6 +28,7 @@ type Case struct {
 	Name       string
 	Dir        string
 	SkipDelete bool
+	Timeout    int
 
 	Client          client.Client
 	DiscoveryClient discovery.DiscoveryInterface
@@ -152,6 +153,7 @@ func (t *Case) LoadTestSteps() error {
 
 	for index, files := range testStepFiles {
 		testStep := &Step{
+			Timeout: t.Timeout,
 			Index:   int(index),
 			Asserts: []runtime.Object{},
 			Apply:   []runtime.Object{},

--- a/pkg/test/harness.go
+++ b/pkg/test/harness.go
@@ -52,6 +52,7 @@ func (h *Harness) LoadTests(dir string) ([]*Case, error) {
 		}
 
 		tests = append(tests, &Case{
+			Timeout:    h.GetTimeout(),
 			Steps:      []*Step{},
 			Name:       file.Name(),
 			Dir:        filepath.Join(dir, file.Name()),
@@ -60,6 +61,15 @@ func (h *Harness) LoadTests(dir string) ([]*Case, error) {
 	}
 
 	return tests, nil
+}
+
+// GetTimeout returns the configured timeout for the test suite.
+func (h *Harness) GetTimeout() int {
+	timeout := 30
+	if h.TestSuite.Timeout != 0 {
+		timeout = 30
+	}
+	return timeout
 }
 
 // RunTestEnv starts a Kubernetes API server and etcd server for use in the

--- a/pkg/test/step.go
+++ b/pkg/test/step.go
@@ -33,7 +33,7 @@ type Step struct {
 	Apply   []runtime.Object
 	Errors  []runtime.Object
 
-	Timeout  int
+	Timeout         int
 	DiscoveryClient discovery.DiscoveryInterface
 	Client          client.Client
 	Logger          testutils.Logger

--- a/pkg/test/step.go
+++ b/pkg/test/step.go
@@ -33,6 +33,7 @@ type Step struct {
 	Apply   []runtime.Object
 	Errors  []runtime.Object
 
+	Timeout  int
 	DiscoveryClient discovery.DiscoveryInterface
 	Client          client.Client
 	Logger          testutils.Logger
@@ -159,7 +160,7 @@ func (s *Step) Create(namespace string) []error {
 
 // GetTimeout gets the timeout defined for the test step.
 func (s *Step) GetTimeout() int {
-	timeout := 10
+	timeout := s.Timeout
 	if s.Assert != nil && s.Assert.Timeout != 0 {
 		timeout = s.Assert.Timeout
 	}

